frontend/components/feature/query-filters/filter-container.tsx

```diff
@@ -22,24 +28,32 @@ export function FilterComponent({ field, aggregation, searchVisible }: { field:
   }
 }
 
-export function FilterContainer({ field, aggregation }: { field: AggregationConfig; aggregation: AggregationConfig["key"] }) {
+type FilterContainerProps = {
+  field: AggregationConfig;
+  aggregation: AggregationConfig['key'];
+  isOpen?: boolean;
+};
+
+export function FilterContainer({ field, isOpen, aggregation }: FilterContainerProps) {
+  const { t } = useI18n();
   const [openItem, setOpenItem] = useState<string | undefined>(undefined);
   const [searchVisible, setSearchVisible] = React.useState(false);
-  const { t } = useI18n();
+
+  useEffect(() => {
+    setOpenItem(isOpen ? field.key : undefined);
+  }, [isOpen]);
```

Comment:
Selon moi ici, tu n'a pas besoin d'utiliser un state variable

Tu pourrais seulement faire ceci:

```suggestion
const openItemKey = isOpen ? field.key : undefined

...

openItemKey === field.key &&

```

Reply:
J'ai besoin de gérer le state interne pour chaque collapse puisse s'ouvrir de façon indépendante. Si je procède comme ça, je dois controller l'était de chaque filtre dans le parent.